frontend/src/routes/Chats/ChatMemberList.tsx

```diff
@@ -0,0 +1,52 @@
+import { useEffect, useState } from "react";
+import { Card, Typography } from "../../components";
+import { ChatMember, User } from "../../types";
+import { socket } from "../../socket";
+
+import ChatMemberItem from "./ChatMemberItem";
+
+type ChatMemberListProps = {
+  members: ChatMember[];
+};
+
+export default function ChatMemberList({ members }: ChatMemberListProps) {
+  const [users, setUsers] = useState<User[]>([]);
+
+  useEffect(() => {
+    const nonDeletedUsers = members
+      .filter((member) => !member.user.deletedAt)
+      .map((member) => member.user);
+
+    setUsers(nonDeletedUsers);
+  }, [members]);
+
+  useEffect(() => {
+    socket.on("userJoined", (user: User) => {
+      setUsers((users) => [...users, user]);
+    });
```

Comment:
Setar apenas como online
```suggestion
  socket.on("userJoined", (user: User) => {
    setUsers((users) =>
      users.map((u) => {
        if (u.id === user.id) {
          return { ...u, online: true };
        }
        return u;
      })
    );
  });
```